addons/html_editor/static/src/fields/html_field.js

```diff
@@ -157,6 +173,8 @@ export class HtmlField extends Component {
             },
             recordInfo: { resModel, resId },
             dropImageAsAttachment: true, // @todo @phoenix always true ?
+            dynamicPlaceholder: this.dynamicPlaceholder,
```

Comment:
This option is not needed if the utility function is called inside the plugin.

In order to call the method from the plugin:
As we currently need the name of the field and the record in the utility function, the utility function can be refactored, renamed and the options can be configured here:
1. rename `useDynamicPlaceholder` with `createDynamicPlaceholder`
2. change signature with something like `createDynamicPlaceholder({el, fieldName, record}) {`

The option for the dynamicPlaceholder can then be:
```js
config = {
  // ...
  dynamicPlaceholder: {
    fieldName: this.props.name,
    record: this.props.record,
    modelReferenceField: this.props.dynamicPlaceholderModelReferenceField
  }
  // ...
}
```